From 446dc61a918683fc7d3cfbf14077b32fe120722b Mon Sep 17 00:00:00 2001
From: Wei Xu <weixugc@google.com>
Date: Tue, 21 Jan 2020 23:30:09 -0800
Subject: [PATCH] Wait for user-data to be available before running
 cloud-init.service.

Because of delay in metadata endpoints becoming active, cloud-init often
ends up running when 'user-data' is not accessible. This is a workaround
until a better signal for metadata reachability is available.

Original Author: Aditya Kali <adityakali@google.com>
Original Date:   Fri Aug 4 14:52:38 2017 -0700
---
 systemd/cloud-init.service.tmpl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/systemd/cloud-init.service.tmpl b/systemd/cloud-init.service.tmpl
index d8dc393e..0a57c31a 100644
--- a/systemd/cloud-init.service.tmpl
+++ b/systemd/cloud-init.service.tmpl
@@ -18,6 +18,8 @@ Before=wicked.service
 After=dbus.service
 {% endif %}
 Before=network-online.target
+After=user-data-online.target
+Wants=user-data-online.target
 {% if variant in ["ubuntu", "unknown", "debian"] %}
 Before=sysinit.target
 Conflicts=shutdown.target
-- 
2.25.0.341.g760bfbb309-goog

